Code/Tools/SerializeContextTools/Application.h

```diff
@@ -11,25 +11,35 @@
 #include <AzToolsFramework/Application/ToolsApplication.h>
 #include <AzCore/IO/Path/Path.h>
 
-namespace AZ
+namespace AZ::IO
 {
-    namespace SerializeContextTools
+    class FileDescriptorCapturer;
+}
+
+namespace AZ::SerializeContextTools
+{
+    class Application final
+        : public AzToolsFramework::ToolsApplication
     {
-        class Application final
-            : public AzToolsFramework::ToolsApplication
-        {
-        public:
-            Application(int argc, char** argv);
-            ~Application() override = default;
-            
-            const char* GetConfigFilePath() const;
-            AZ::ComponentTypeList GetRequiredSystemComponents() const override;
-            void QueryApplicationType(AZ::ApplicationTypeQuery& appType) const override;
-
-        protected:
-            void SetSettingsRegistrySpecializations(AZ::SettingsRegistryInterface::Specializations& specializations) override;
-
-            AZ::IO::FixedMaxPath m_configFilePath;
-        };
-    } // namespace SerializeContextTools
-} // namespace AZ
+    public:
+        Application(int argc, char** argv);
+        Application(int argc, char** argv, AZ::IO::FileDescriptorCapturer* stdoutCapturer);
```

Comment:
Can we make the `stdoutCapturer` default to `nullptr` and just have one constructor?

Reply:
Normally you can't if this was a va_list, but its not, so it should be okay?

Reply:
Yes, that can be done